include/nbl/system/ISystem.h

```diff
@@ -197,23 +198,10 @@ class ISystem final : public core::IReferenceCounted
     // @sadiuk add more methods taken from IFileSystem and IOSOperator
     // and implement via m_dispatcher and ISystemCaller if needed
     // (any system calls should take place in ISystemCaller which is called by CAsyncQueue and nothing else)
-    core::smart_refctd_ptr<IFile> createAndOpenFile(const std::filesystem::path& filename)
-    {
-        future_t<core::smart_refctd_ptr<IFile>> future;
-        if (!createFile(future, filename, IFile::ECF_READ))
-            return nullptr;
-
-        return future.get();
-    }
-    core::smart_refctd_ptr<IFile> createAndWriteFile(const std::filesystem::path& filename)
-    {
-        future_t<core::smart_refctd_ptr<IFile>> future;
-        if (!createFile(future, filename, IFile::ECF_WRITE))
-            return nullptr;
-
-        return future.get();
-    }
 
+    virtual void seek(IFile* file, uint32_t pos) = 0;
+    virtual void read(IFile* file, char* outData, size_t count) const;
+    virtual size_t getPos(IFile* file) const = 0;
```

Comment:
no, no, no 

bad API (multithreaded operation on the same file impossible)

We've already established with @Crisspl (see other PR) that files should not be keeping positions as intermediate state, the read and write calls should take absolute offsets into the file.

Reply:
Yeah i moved them there cause criss said we probably should have those functions somewhere in the future. Not using them though.

Reply:
wait what, no, you must have misunderstood something. We dont want any of this in public API, we want to have file access API that allows access like memory (`read(offset,size)`, `write(offset,size)`)

Reply:
basically like OpenGL/Vulkan copy/update buffer sub data but for a file